runtime/compiler/control/OptionsPostRestore.cpp

```diff
@@ -349,63 +400,202 @@ J9::OptionsPostRestore::filterMethod(TR_J9VMBase *fej9, J9Method *method)
+      {
+      TR_ASSERT_FATAL(vLogFileName,
+                      "vlogFileName cannot be NULL if _oldVLogFileName (%s) is not NULL\n",
+                      _oldVLogFileName);
+
+      TR_ASSERT_FATAL(_privateConfig->vLogFile,
+                      "_privateConfig->vLogFile should not be NULL if _oldVLogFileName (%s) is not NULL\n",
+                      _oldVLogFileName);
+
+      j9jit_fclose(_privateConfig->vLogFile);
+      TR::Options::jitPersistentFree(_oldVLogFileName);
+      _oldVLogFileName = NULL;
+      }
+
+   _privateConfig->vLogFile = fileOpen(TR::Options::getCmdLineOptions(), _jitConfig, vLogFileName, "wb", true);
+   TR::Options::setVerboseOptions(_privateConfig->verboseFlags);
+
+   TR_VerboseLog::vlogRelease();
+   }
+
+void
+J9::OptionsPostRestore::openNewRTLog(char *rtLogFileName)
+   {
+   bool closeOldFile = (_oldRtLogFileName != NULL);
+
+   JITRT_LOCK_LOG(_jitConfig);
 
-   J9ClassWalkState classWalkState;
-   J9Class *j9clazz = javaVM->internalVMFunctions->allClassesStartDo(&classWalkState, javaVM, NULL);
-   while (j9clazz)
+   if (closeOldFile)
       {
-      TR_OpaqueClassBlock *clazz = reinterpret_cast<TR_OpaqueClassBlock *>(j9clazz);
-      uint32_t numMethods = fej9->getNumMethods(clazz);
-      J9Method *ramMethods = reinterpret_cast<J9Method *>(fej9->getMethods(clazz));
+      TR_ASSERT_FATAL(rtLogFileName,
+                      "rtLogFileName cannot be NULL if _oldRtLogFileName (%s) is not NULL\n",
+                      _oldRtLogFileName);
 
-      for (uint32_t index = 0; index < numMethods; index++)
+      TR_ASSERT_FATAL(_privateConfig->rtLogFile,
+                      "_privateConfig->rtLogFile should not be NULL if _oldRtLogFileName (%s) is not NULL\n",
+                      _oldRtLogFileName);
+
+      j9jit_fclose(_privateConfig->rtLogFile);
+      TR::Options::jitPersistentFree(_oldRtLogFileName);
+      _oldRtLogFileName = NULL;
+      }
+
+   _privateConfig->rtLogFile = fileOpen(TR::Options::getCmdLineOptions(), _jitConfig, rtLogFileName, "wb", true);
+
+   JITRT_UNLOCK_LOG(_jitConfig);
+
+   TR::CompilationInfoPerThread * const * arrayOfCompInfoPT = _compInfo->getArrayOfCompilationInfoPerThread();
+   for (int32_t i = _compInfo->getFirstCompThreadID(); i <= _compInfo->getLastCompThreadID(); i++)
+      {
+      TR::CompilationInfoPerThread *compThreadInfoPT = arrayOfCompInfoPT[i];
+      if (closeOldFile)
          {
-         filterMethod(fej9, &ramMethods[index]);
+         compThreadInfoPT->closeRTLogFile();
          }
+      compThreadInfoPT->openRTLogFile();
+      }
+   }
+
+void
+J9::OptionsPostRestore::openLogFilesIfNeeded()
+   {
+   _privateConfig->vLogFileName = _jitConfig->vLogFileName;
+
+   char *vLogFileName = _privateConfig->vLogFileName;
+   char *rtLogFileName = _privateConfig->rtLogFileName;
+
+   // if _oldVLogFileName != vLogFileName, it is not possible that
+   // _oldVLogFileName != NULL && vLogFileName == NULL
+   if (_oldVLogFileName != vLogFileName)
```

Comment:
Here, it being the same means someone spcified `-Xjit:vlog=vlog` in the JVM command line at startup AND also in the post-restore options. If they didn't specify it in post restore options, then `vlogFileName` will be NULL.

The reason I didn't attempt to try to append to it is because if the restored process' PID is different, which I've seen can happen, the suffix of the vlog will be different from the one pre-checkpoint.